### PR TITLE
[LIBRARY_REQUEST] Make CCCL library available for C++ as well

### PR DESCRIPTION
**Pull request by bernhardmgruber**
The CCCL (CUDA Core Compute Libraries) are currently only availabel for the CUDA language. However, they also contain many parts that compile just fine with plain C++.

This PR makes CCCL available for the C++ language as well. As a drive-by fix the recent name change from CUDA C++ Core Libraries to CUDA Core Compute Libraries is appiled (we also ship Python libraries now).

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -4050,6 +4050,25 @@ libs.catch2.versions.301.version=3.0.1
...
+libs.cccl.versions.260.version=2.6.0
+libs.cccl.versions.260.path=/opt/compiler-explorer/libs/cccl/2.6.0/thrust:/opt/compiler-explorer/libs/cccl/2.6.0/cub:/opt/compiler-explorer/libs/cccl/2.6.0/libcudacxx/include
+libs.cccl.versions.250.version=2.5.0
+libs.cccl.versions.250.path=/opt/compiler-explorer/libs/cccl/2.5.0/thrust:/opt/compiler-explorer/libs/cccl/2.5.0/cub:/opt/compiler-explorer/libs/cccl/2.5.0/libcudacxx/include
+libs.cccl.versions.240.version=2.4.0
+libs.cccl.versions.240.path=/opt/compiler-explorer/libs/cccl/2.4.0/thrust:/opt/compiler-explorer/libs/cccl/2.4.0/cub:/opt/compiler-explorer/libs/cccl/2.4.0/libcudacxx/include
+libs.cccl.versions.230.version=2.3.0
+libs.cccl.versions.230.path=/opt/compiler-explorer/libs/cccl/2.3.0/thrust:/opt/compiler-explorer/libs/cccl/2.3.0/cub:/opt/compiler-explorer/libs/cccl/2.3.0/libcudacxx/include
+libs.cccl.versions.220.version=2.2.0
+libs.cccl.versions.220.path=/opt/compiler-explorer/libs/cccl/2.2.0/thrust:/opt/compiler-explorer/libs/cccl/2.2.0/cub:/opt/compiler-explorer/libs/cccl/2.2.0/libcudacxx/include
```
Comment (bernhardmgruber) [inline etc/config/c++.amazon.properties:4070]: This is a plain copy from `cuda.amazon.properties`

**Review by junlarsen — APPROVED** [review]

File: etc/config/c++.amazon.properties [hunk 2]
```diff
@@ -4050,6 +4050,25 @@ libs.catch2.versions.301.version=3.0.1
 libs.catch2.versions.301.path=/opt/compiler-explorer/libs/catch2/v3.0.1/src
 libs.catch2.versions.301.staticliblink=Catch2d
 
+libs.cccl.name=CUDA Core Compute Libraries
```
Comment (ericniebler) [inline etc/config/c++.amazon.properties:4053]:
```suggestion
libs.cccl.name=CUDA Core Compute Libraries (CCCL)
```

Comment (bernhardmgruber) [reply etc/config/c++.amazon.properties:4053]: Thx for the suggestion! Applied.

**Comment by junlarsen** [comment]
Thanks!

**Pull request merged**